### PR TITLE
SOV-1865: Added subscription payload to notifications

### PR DESCRIPTION
**Pull request by BetsyBraddock**
https://sovryn.atlassian.net/browse/SOV-1865

### CODE REVIEW

File: src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx [hunk 1]
```diff
@@ -45,6 +46,14 @@ export const NotificationSettingsDialog: React.FC<INotificationSettingsDialogPro
   const [isDiscordActive, setIsDiscordActive] = useState(false);
 
   const emailIsValid = useMemo(() => !email || validateEmail(email), [email]);
+  const isSubscribedToD1Notifications = (subscriptions: Subscription[]) => {
```
Comment (soulBit) [inline src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx:49]: You should be able to extract this function to outside of this component definition (so above line 26), currently it redefines new reference to this function every time component renders - not a big thing, but good practice.

File: src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx [hunk 2]
```diff
@@ -76,6 +85,33 @@ export const NotificationSettingsDialog: React.FC<INotificationSettingsDialogPro
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [notificationUser]);
 
+  const updateSubscriptions = () => {
```
Comment (soulBit) [inline src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx:88]: Wrap this function in `useCallback()` with `notificationUser` in dependency array

**Review by soulBit — CHANGES_REQUESTED** [review]

File: src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx (hunk 2 above)
Comment (BetsyBraddock) [reply src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx:88]: I don't think the result of this should be cached - this function is only called when the user actually submits the form. If notificationUser is the dependency array, then when the user submits the form the notificationUser is not updated and the form shows the old version

Comment (BetsyBraddock) [reply src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx:88]: Thanks soulbit! Making this change now

File: src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx [hunk 3]
```diff
@@ -76,6 +91,33 @@ export const NotificationSettingsDialog: React.FC<INotificationSettingsDialogPro
...
+    if (notificationUser) {
+      const subscriptions: Subscription[] = JSON.parse(
+        JSON.stringify(notificationUser?.subscriptions),
+      );
+      const d1Notifications = Object.values(Notification);
+      d1Notifications.forEach(item => {
+        const foundSubscription = subscriptions.findIndex(
+          sub => sub.notification === item,
+        );
+        if (foundSubscription) {
```
Comment (soulBit) [inline src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx:104]:
```suggestion
        if (foundSubscription >= 0) {
```
Currently crashes if you try to enable notifications with a new wallet as `foundSubscription === -1`, so next line fails

**Review by soulBit — CHANGES_REQUESTED** [review]
Applied changes look good, and functionality works as specified in story. Just one last fix to make

**Review by soulBit — APPROVED** [review]

**Pull request merged**